File: SecondYear/Serve-er_SarveshBaranwal_2226cseAI&ml1045_2/Week_4/traffic.cpp
int traffic(int n, int m, vector<int> vehicle) {

    int i = 0, j = 0, ans = 1, sums = 0, k = m;

    while(j < n){

        if (k <= 0 && i < j && vehicle[j] == 0) {

            sums--;

            if(vehicle[i] == 0){

                k++;

            }

            i++;

            continue;

        }

        

        if(vehicle[j] == 0) k--;

        sums++;

        j++;

        ans = max(ans, sums);

    }

    return ans;

}

File: SecondYear/Anantj2499_AnantJaiswal_2226cseml1088_2/Week_4/Traffic.java
public class Traffic {
    public static int traffic(int n, int m, int []vehicle) {
        // Write your code here.
        int maxL = 0,left = 0,right = 0,sum=0;
        while(right<n){
            sum += vehicle[right];
            // here checking the rule
            if(sum<right-left+1-m){
                sum -= vehicle[left];
                left ++;
            }
            maxL = Math.max(maxL, right - left + 1);
            right++;
        }
        return maxL;
    }
}